### PR TITLE
fix: validate admin console password length

### PR DESCRIPTION
**Pull request by JGAntunes**
#### What this PR does / why we need it:
Make sure we validate the admin console password length requiring it to be at least 6 characters, similar to what already [happens in KOTS](https://github.com/replicatedhq/kots/blob/dab9434bb24c13ffb5f37b528c515121b31db089/pkg/util/password_prompt.go#L22-L25)

#### Which issue(s) this PR fixes:

https://app.shortcut.com/replicated/story/104936/admin-console-minimum-password-length-inconsistent

#### Does this PR require a test?

I've added a test to the password validation function and was trying to see if there's a way that we could test the `maybeAskAdminConsolePassword` method, but since it relies on input prompts I'm unsure what would be the best mechanism to do so 🤔 open to suggestions.

Meanwhile I've also run a couple of manual tests in the local dev env:
```
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install --license local-dev/Joao.yaml
? Set the Admin Console password: ***
? Confirm the Admin Console password: ***
Passwords must have more than 6 characters. Please try again.
? Set the Admin Console password: ******
? Confirm the Admin Console password: ******
Passwords don't match. Please try again.
? Set the Admin Console password: ******
? Confirm the Admin Console password: ******
✔  Host files materialized!
(...)
```

```
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install --license local-dev/Joao.yaml --no-prompt
The Admin Console password is set to password
(...)
```

```
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install --license local-dev/Joao.yaml --admin-console-password 123
Passwords must have more than 6 characters. Please try again.
unable to set the Admin Console password
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install --license local-dev/Joao.yaml --admin-console-password 123456
✔  Host files materialized!
(...)
```

```
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install help
NAME:
   embedded-cluster install - Install embedded-cluster

USAGE:
   embedded-cluster install command [command options]

COMMANDS:
   help, h  Shows a list of commands or help for one command

OPTIONS:
   --admin-console-password value             Password for the Admin Console (minimum 6 characters)
   (...)
```

```
root@node0:/replicatedhq/embedded-cluster# output/bin/embedded-cluster install --license local-dev/Joao.yaml
? Set the Admin Console password (minimum 6 characters):
unable to ask for input: interrupt
```

#### Does this PR require a release note?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: cmd/embedded-cluster/install.go [hunk 1]
```diff
@@ -551,31 +554,42 @@ func runOutro(c *cli.Context, applier *addons.Applier, cfg *k0sconfig.ClusterCon
 func maybeAskAdminConsolePassword(c *cli.Context) (string, error) {
 	defaultPassword := "password"
 	userProvidedPassword := c.String("admin-console-password")
-	if c.Bool("no-prompt") {
-		if userProvidedPassword != "" {
-			return userProvidedPassword, nil
-		} else {
```
Comment (JGAntunes) [inline cmd/embedded-cluster/install.go:557]: We always prioritised the user provided password whether we have `no-prompt` or not.

File: cmd/embedded-cluster/install.go [hunk 2]
```diff
@@ -551,31 +554,42 @@ func runOutro(c *cli.Context, applier *addons.Applier, cfg *k0sconfig.ClusterCon
...
-			logrus.Infof("The Admin Console password is set to %s", defaultPassword)
-			return defaultPassword, nil
-		}
-	}
+	// If there's a user provided password we'll try that first
 	if userProvidedPassword != "" {
+		if !validateAdminConsolePassword(userProvidedPassword, "", false) {
+			return "", fmt.Errorf("unable to set the Admin Console password")
+		}
```
Comment (JGAntunes) [inline cmd/embedded-cluster/install.go:561]: I'm unsure if we only wanna validate the prompts on the case where the user provides a password through a CLI param. I assumed the latter.

File: cmd/embedded-cluster/install.go [hunk 3]
```diff
@@ -551,31 +554,42 @@ func runOutro(c *cli.Context, applier *addons.Applier, cfg *k0sconfig.ClusterCon
...
-			// TODO: Should we add extra password validation here? e.g length, complexity etc
+		if validateAdminConsolePassword(promptA, promptB, true) {
 			return promptA, nil
 		}
-		logrus.Info("Passwords don't match. Please try again.")
 	}
 	return "", fmt.Errorf("unable to set the Admin Console password after %d tries", maxTries)
 }
 
+func validateAdminConsolePassword(password, passwordCheck string, mustMatch bool) bool {
```
Comment (laverya) [inline cmd/embedded-cluster/install.go:581]: The only time this is called with `mustMatch: false` is when it's provided via the CLI. Would it make more sense for that function call to just supply the password twice to the function?

File: cmd/embedded-cluster/install_test.go [hunk 4]
```diff
@@ -200,3 +200,70 @@ versionLabel: testversion
 		})
 	}
 }
+
+func Test_validateAdminConsolePassword(t *testing.T) {
+
+	tests := []struct {
```
[the remark this answers is not in the document]
Comment (laverya) [reply cmd/embedded-cluster/install_test.go:206]: we have the ability to run integration tests with `expect`, but that is probably a bit farther than we want to go here

File: cmd/embedded-cluster/install.go (hunk 3 above)
Comment (JGAntunes) [reply cmd/embedded-cluster/install.go:581]: Personally, I prefer being explicit about the functionality VS having it be dependent on behaviour which is non-explicit. But happy to change it if y'all think differently.

File: cmd/embedded-cluster/install_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (JGAntunes) [reply cmd/embedded-cluster/install_test.go:209]: I've added a simple set of tests to `maybeAskAdminConsolePassword` without exercising the prompts.

File: cmd/embedded-cluster/install.go (hunk 3 above)
Comment (laverya) [reply cmd/embedded-cluster/install.go:581]: my feeling is that 'mustMatch' isn't particularly explicit - if we're going to do something like that, we should have two functions one for 'passwordLength' and another for 'passwordsMatch' IMO

@sgalsaleh ?

Comment (JGAntunes) [reply cmd/embedded-cluster/install.go]: Went with your original suggestion via - https://github.com/replicatedhq/embedded-cluster/pull/1262/commits/9ab78499e3eefeed87cf8bb11998303c503ef779 - let me know what you think @laverya

**Review by laverya — APPROVED** [review]

**Pull request merged**